src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,79 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useState } from 'react';
+import { Filter } from '../../types/EnumFilter';
+import { FiltersType } from '../../types/Filter';
+
+type Props = {
+  getChangeFilters: (filter: FiltersType) => void;
+  filters: FiltersType;
+};
+
+export const TodoFilter: React.FC<Props> = ({ getChangeFilters, filters }) => {
+  const [clearInput, setClearInput] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const handleFiltersChange = (optionValue: Filter, inputQuery: string) => {
+    setInputValue(inputQuery);
+    setClearInput(true);
+    getChangeFilters({
+      option: optionValue as Filter,
```

Comment:
Type Assertion: The use of `as Filter` is a type assertion. Ensure that the `optionValue` is validated or inferred correctly to avoid potential runtime errors.